tc/core/polyhedral/memory_promotion.cc

```diff
@@ -393,13 +459,21 @@ isl::multi_aff TensorReferenceGroup::promotion() const {
   isl::map map = scopedAccesses();
   auto accessSpace = map.get_space();
 
-  // lower bounds space is S -> P; which we transform into [S -> O] -> P
-  auto lowerBounds = approximation.lowerBounds().pullback(
-      isl::multi_aff::domain_map(accessSpace));
-  auto promotion = isl::multi_aff::range_map(accessSpace)
-                       .reset_tuple_id(isl::dim_type::out) -
-      lowerBounds;
-  return promotion;
+  // Construct a projection multi-aff in [S -> O] -> S
+  // for futher precomposition.
+  auto originalSpaceInserter = isl::multi_aff::domain_map(accessSpace);
+
+  // Lower bounds and offsets space is S -> P; transform into [S -> O] -> P.
+  auto lowerBounds =
+      approximation.lowerBounds().pullback(originalSpaceInserter);
+  auto offsets = approximation.offsets().pullback(originalSpaceInserter);
+
+  // Create identity mapping from original to promoted,
+  // resetting the tuple name for the promoted tensor.
```

Comment:
Why does the tuple name need to be reset?
If you want to make as distinction between original and promoted tensor,
then it's better to assign an explicit name than to simply remove it.

Reply:
That names is not yet known at this point.